### PR TITLE
[MRG] Rounding with raw.time_as_index()

### PR DESCRIPTION
**Pull request by wronk**
This adds an option to round the indices returned by raw.time_as_index(). The original formulation truncates decimals after it calculates an index, which can lead to non-unique indices when doing: `raw.time_as_index(raw.times)`

### CODE REVIEW

**Comment by wronk** [comment]
From issue #2311

File: mne/io/base.py [hunk 1]
```diff
@@ -1702,25 +1705,43 @@ def _allocate_data(data, data_buffer, data_shape, dtype):
...
+    np.around will return the nearest even number for values exactly between
+        two integers.
     """
     index = np.atleast_1d(times) * sfreq
     index -= (first_samp if use_first_samp else 0)
-    return index.astype(int)
+
+    # Round or truncate time indices
+    if use_rounding:
+        return np.around(index).astype(int)
```
Comment (agramfort) [inline mne/io/base.py]: np.around is better than np.round?

**Comment by agramfort** [comment]
besides LGTM

File: mne/io/base.py (hunk 1 above)
Comment (wronk) [inline mne/io/base.py]: They're the same function, but I can easily change to `np.round` if it's clearer.

Comment (Eric89GXL) [inline mne/io/base.py]: I'd change it, people won't have to wonder if it's the same then

**Comment by agramfort** [comment]
thanks @wronk

**Pull request merged**